iis/datadog_checks/iis/iis.py

```diff
@@ -55,62 +55,71 @@ def get_iishost(self, instance):
         return "iis_host:{}".format(self.normalize(iis_host))
 
     def check(self, instance):
-        sites = instance.get('sites')
-        if sites is None:
-            expected_sites = set()
-        else:
-            expected_sites = set(sites)
+        sites = instance.get('sites', [])
+        expected_sites = set(sites)
         # _Total should always be in the list of expected sites; we always
         # report _Total
         expected_sites.add(TOTAL_SITE)
 
         self.log.debug("Expected sites is {}".format(expected_sites))
-        key = hash_mutable(instance)
-        for inst_name, dd_name, metric_func, counter in self._metrics[key]:
+        instance_hash = hash_mutable(instance)
+        for inst_name, dd_name, metric_func, counter in self._metrics[instance_hash]:
             try:
-                try:
-                    vals = counter.get_all_values()
-                except Exception as e:
-                    self.log.error("Failed to get_all_values {} {}: {}".format(inst_name, dd_name, e))
-                    continue
-
-                for sitename, val in iteritems(vals):
-                    tags = []
-                    if key in self._tags:
-                        tags = list(self._tags[key])
-                    tags.append(self.get_iishost(instance))
-
-                    try:
-                        if not counter.is_single_instance():
-                            if sites and sitename != TOTAL_SITE and sitename not in sites:
-                                continue
```

Comment:
moved this check & continue to the top of the for loop, there is no point in start constructing tags only to skip the loop interation a few lines later